File: hidrive_cctv_monitoring/ios/ios-plugins/HiDriveProxyPlugin.swift
import Foundation
import Capacitor

@objc(HiDriveProxyPlugin)
public class HiDriveProxyPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier   = "HiDriveProxyPlugin"
    public let jsName       = "HiDriveProxy"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "start", returnType: CAPPluginReturnPromise),
    ]

    private var server: HiDriveProxyServer?

    @objc func start(_ call: CAPPluginCall) {
        guard let username = call.getString("username"),
              let password = call.getString("password") else {
            call.reject("username and password are required")
            return
        }
        let port = UInt16(call.getInt("port") ?? 18765)

        let srv = HiDriveProxyServer(port: port, username: username, password: password)
        do {
            try srv.start()
            self.server = srv
            call.resolve(["port": port])
        } catch {
            call.reject(error.localizedDescription)
        }
    }
}

File: hidrive_cctv_monitoring/android/android/app/src/main/java/ai/hidrive/cctv/HiDriveProxyPlugin.java
package ai.hidrive.cctv;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "HiDriveProxy")
public class HiDriveProxyPlugin extends Plugin {
    private HiDriveProxyServer server;

    @PluginMethod
    public void start(PluginCall call) {
        String username = call.getString("username", "");
        String password = call.getString("password", "");
        int port = call.getInt("port", 18765);

        if (server != null) server.stop();

        server = new HiDriveProxyServer(port, username, password);
        try {
            server.start();
            JSObject ret = new JSObject();
            ret.put("port", port);
            call.resolve(ret);
        } catch (Exception e) {
            call.reject("Failed to start proxy: " + e.getMessage());
        }
    }

    @PluginMethod
    public void stop(PluginCall call) {
        if (server != null) { server.stop(); server = null; }
        call.resolve();
    }

    @Override
    protected void handleOnDestroy() {
        if (server != null) server.stop();
    }
}
